src/Workspaces/Remote/Core/ProjectSystem/IWorkspaceProject.cs

```diff
@@ -0,0 +1,38 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Collections.Generic;
+using System.Threading;
+using System.Threading.Tasks;
+using StreamJsonRpc;
+
+namespace Microsoft.CodeAnalysis.Remote.ProjectSystem
+{
+    [RpcMarshalable]
+    public interface IWorkspaceProject : IAsyncDisposable
+    {
+        Task SetDisplayNameAsync(string displayName, CancellationToken cancellationToken);
+
+        Task SetCommandLineArgumentsAsync(IReadOnlyList<string> arguments, CancellationToken cancellationToken);
+        Task SetBuildSystemPropertiesAsync(IReadOnlyDictionary<string, string> properties, CancellationToken cancellationToken);
+
+        Task AddSourceFilesAsync(IReadOnlyList<SourceFileInfo> sourceFiles, CancellationToken cancellationToken);
+        Task RemoveSourceFilesAsync(IReadOnlyCollection<string> sourceFiles, CancellationToken cancellationToken);
+
+        Task AddMetadataReferencesAsync(IReadOnlyCollection<MetadataReferenceInfo> metadataReferences, CancellationToken cancellationToken);
```

Comment:
why IROL in some cases, and IROC in others?

Reply:
Was making a subtle difference between "order matters" and "order doesn't" but I should just not do that.

Reply:
Everything is IReadOnlyList now.